arangod/Aql/EnumerateCollectionExecutor.cpp

```diff
@@ -48,22 +49,20 @@
 using namespace arangodb;
 using namespace arangodb::aql;
 
-namespace {
-std::vector<size_t> const emptyAttributePositions;
-}
-
 EnumerateCollectionExecutorInfos::EnumerateCollectionExecutorInfos(
     RegisterId outputRegister, aql::QueryContext& query,
     Collection const* collection, Variable const* outVariable,
     bool produceResult, Expression* filter,
-    arangodb::aql::Projections projections, bool random, bool count,
-    ReadOwnWrites readOwnWrites)
+    arangodb::aql::Projections projections,
+    std::vector<std::pair<VariableId, RegisterId>> filterVarsToRegs,
```

Comment:
do we intentionally copy filterVarsToRegs?

Reply:
`filterVarsToRegs` are a local variable created in `EnumerateCollectionNode::createBlock(...)` and moved there into the `EnumerateCollectionExecutorInfos` constructor. In the constructor finally, they are also moved into the `_filterVarsToRegs` member, so we are not copying here, only moving.